### PR TITLE
doc: update tests-integration-testing-teuthology-workflow.rst

### PR DESCRIPTION
**Pull request by VallariAg**
1. Add "Infrastructure" section.
2. Move "Naming the ceph-ci branch" section under "Getting binaries - Build Ceph". Also mention about centos9-only trick.
3. In "Teuthology Archives", mention about developer playground machines and log files.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst [hunk 1]
```diff
@@ -3,11 +3,44 @@
...
--------------------
-
-Getting binaries
-****************
+Infrastructure
+--------------
+
+Components:
+
+1. `ceph-ci`_: Clone of main ceph repository, used for triggering jenkins ceph builds for development.
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst:11]: Of the main

Ceph capital

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst [hunk 2]
```diff
@@ -3,11 +3,44 @@
...
+2. `ceph_jenkins`_: It is responsible for triggering build, uploading packages to Chacra, and pushing updates about the build to Shaman.
+3. `Shaman`_: UI Interface used to check build statuses. In it's backend, it is a REST API that talks to jenkins.
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst:13]: status
its
Jenkins

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst [hunk 3]
```diff
@@ -3,11 +3,44 @@
...
+4. `Chacra`_: Service where packages are uploaded. The binaries uploaded here can be downloaded and used by anyone.
+5. Teuthology CLI: Developers can use various teuthology commands to schedule and manage test runs. 
+6. Teuthology: This is the component responsible for pushing test jobs to beanstalk queue and 
+   paddles. It is also responsible for picking jobs from queue and running tests. Developers usually don't interact with it. 
+7. Beanstalk queue: A priority queue with all the queued jobs. Developers usually don't interact with it. 
+8. Paddles: Backend service which stores all test run information. Developers usually don't need to interact with it.
+9.  `Pulpito`_: UI Interface (of information stored on paddles) used by developers to see full information about their scheduled tests,
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst:20]: Extra space

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst [hunk 4]
```diff
@@ -3,11 +3,44 @@
...
+8. Paddles: Backend service which stores all test run information. Developers usually don't need to interact with it.
+9.  `Pulpito`_: UI Interface (of information stored on paddles) used by developers to see full information about their scheduled tests, 
+   along with it's status and results. 
+10. Testnodes (`smithi`_): A cluster of various types of machines that are used for running tests.
+   Developers usually schedule tests to be run on "smithi" machines which are dedicated test nodes for teuthology integration testing.
+
+Each teuthology test *run* contains multiple test *jobs*. Each job runs in an environment isolated from other jobs, on a different collection of test nodes.
+
+
+To test a change in ceph, follow:
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-workflow.rst:28]: Ceph

**Review by zdover23 — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/59548 - Squid backport
https://github.com/ceph/ceph/pull/59549 - Reef backport
https://github.com/ceph/ceph/pull/59550 - Quincy backport

**Pull request merged**